### PR TITLE
Pr101x L1T Kalman Filter BMTF emulator and DataFormat

### PR DESCRIPTION
**Pull request by rekovic**
New L1T Kalman Filter Muon algorithm in BMTF.  

Emulator packated in L1Trigger/L1TMuonBarrel.

New DataFormats
- L1MuKBMTCombinedStub
    - Made of TwinMux eta and phi segments
- L1MuKBMTrack
     - for KalmanFilter Track Finder

Updated DataFormats
- RegionalMuonCand  (update)
    - addition of a second hwPt and a hwDXY in the-> zero values by default for OMTF/EMTF
    - DataFormat is needed for the unpacker development as we are going to take data with
this algo in the firmware

Note:
To run include simKBmtfStubs*simKBmtfDigis in EmuL1Simulation sequence.
As is should not have any effect.

### CODE REVIEW

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
please test

File: L1Trigger/L1TMuonBarrel/interface/L1TMuonBarrelKalmanLUTs.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#ifndef __L1TMuonBarelKalmanLUTs_h
```
Comment (thomreis) [inline L1Trigger/L1TMuonBarrel/interface/L1TMuonBarrelKalmanLUTs.h]: Please correct the typo here. Barel -> Barrel

File: L1Trigger/L1TMuonBarrel/plugins/L1TMuonBarrelKalmanTrackProducer.cc [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+   private:
+      void beginStream(edm::StreamID) override;
+      void produce(edm::Event&, const edm::EventSetup&) override;
+      void endStream() override;
+  edm::EDGetTokenT<std::vector<L1MuKBMTCombinedStub> > src_;
+  std::vector<int> bx_;
+  L1TMuonBarrelKalmanAlgo *algo_;
```
Comment (thomreis) [inline L1Trigger/L1TMuonBarrel/plugins/L1TMuonBarrelKalmanTrackProducer.cc]: Better to use smart pointers.

File: L1Trigger/L1TMuonBarrel/plugins/L1TMuonBarrelKalmanStubProducer.cc [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+   private:
+      void beginStream(edm::StreamID) override;
+      void produce(edm::Event&, const edm::EventSetup&) override;
+      void endStream() override;
+  edm::EDGetTokenT<L1MuDTChambPhContainer> srcPhi_;
+  edm::EDGetTokenT<L1MuDTChambThContainer> srcTheta_;
+  L1TMuonBarrelKalmanStubProcessor * proc_;
```
Comment (thomreis) [inline L1Trigger/L1TMuonBarrel/plugins/L1TMuonBarrelKalmanStubProducer.cc]: Better to use a smart pointer.

File: L1Trigger/L1TMuonBarrel/interface/L1TMuonBarrelKalmanAlgo.h [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  void estimateChiSquare(L1MuKBMTrack&);
+  int rank(const L1MuKBMTrack&);
+  int wrapAround(int,int);
+  std::pair<bool,uint> getByCode(const L1MuKBMTrackCollection& tracks,int mask);
+  std::map<int,int> trackAddress(const L1MuKBMTrack&,int&);
+  int encode(bool ownwheel,int sector,bool tag); 
+  uint twosCompToBits(int);
+
+  //LUT service
+  L1TMuonBarrelKalmanLUTs* lutService_;
```
Comment (thomreis) [inline L1Trigger/L1TMuonBarrel/interface/L1TMuonBarrelKalmanAlgo.h]: Better use a smart pointer.

**Review by thomreis — COMMENTED** [review]
There are couple of places where bare pointers could probably be replaced by smart pointers.

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
+1

**Comment by fabiocos** [comment]
+1 

needed for commissioning DQM, no effect in standard workflows

**Pull request merged**